README.md

```diff
@@ -114,6 +114,9 @@ The following provides information for developers looking to maintain or extend
 
 ## Architecture
 
+## Pre-commit hooks
+Git hooks are used to run formatting (isort, black), linting (flake8), and type-checking (mypy). You can install the pre-commit scripts with `pre-commit install`. Thereafter, the scripts defined in `.pre-commit-config.yaml` will be run on the files included in a commit. Specific tool configurations are included in `pyproject.toml`.
```

Comment:
This is useful. Worth mentioning that `pre-commit install` is run by `scripts/setup` too